install_config/upgrading/blue_green_deployments.adoc

```diff
@@ -0,0 +1,157 @@
+== Preparing for Upgrade
+
+xref:../../install_config/upgrading/manual_upgrades.adoc#upgrading-masters[After
+the master and etcd servers have been upgraded], you need to ensure that your
+current production nodes are labeled either blue or green. In this example, the
+current installation will be blue and the new environment will be green.
+
+----
+$ oc label --all nodes color=blue
+----
+
+In the case of nodes requiring the uptime guarantees of a blue-green deployment,
+the `-l` flag can be used to match a subset of the environment using a selector.
+
+Next, xref:../install/advanced_install.adoc#adding-nodes-advanced[the green
+environment] should be created for any nodes that are to be replaced. Ansible
+can apply the `*color=green*` using the `*openshift_node_labels*` variable for
+each node.
+
+In order to delay workload scheduling until the nodes are
+xref:../../dev_guide/application_health.adoc#dev-guide-application-health[healthy],
+be sure to set the `*openshift_schedulable=false*` variable.
+
+Once the green nodes are in *Ready* state, they can be made schedulable. Then,
+blue nodes are disabled so that no new pods are run on them.
+
+----
+# oadm manage-node --schedulable=true --selector=color=green
+# oadm manage-node --schedulable=false --selector=color=blue
+----
+
+A common practice is to scale the registry and router pods until they are
+migrated to the green nodes. For these pods, a _canary_ deployment approach is
+commonly used. Scaling them up will make them immediately active on the new
+nodes. Pointing the deployment configuration to the new image initiates a
+rolling update. However, because of node anti-affinity, and the fact that the
+blue nodes are still unschedulable, the deployments to the old nodes will fail.
+At this point, the registry and router deployments can be scaled down to the
+original number of pods. At any given point, the original number of pods is
+still available so no capacity is lost.
+
+[[manual-upgrades-warming-the-new-nodes]]
+== Warming the New Nodes
+
+In order for pods to be migrated from the blue environment to the green, the
+images must be pulled. Network latency and load on the registry can cause delays
+if there is not sufficient capacity built in to the environment. Often, the best
+way to minimize impact to the running system is to trigger new pod deployments that
+will land on the new nodes. Accomplish this by importing new *ImageStreams*.
+
+A major release of {product-title} is the motivation for a
+blue-green deployment. At that time, new *ImageStreams* become
+available for users of Source-to-Image. Upon import, any builds or deployments
+configured with *ImageChangeTriggers* are automatically created.
+
+It is important to realize that this process can trigger a large number of
+builds. The good news is that the builds are performed on the green nodes and,
+therefore, do not impact any traffic on the blue deployment.
+
+Monitor build progress:
```

Comment:
Or even "To monitor build progress across all namespaces (projects) in the cluster:"